File: problems/chapter06/nicetak/abc026_d.jl
function main()
    A, B, C = parse.(Float64, split(readline()))
    g(t; A=A, B=B, C=C) = A * t + B * sin(C * t * π) - 100
    t̲, t̄ = 0, (B + 100) / A
    ε = 1e-6

    t = (t̄ + t̲) / 2
    while abs(g(t)) > ε
        if g(t) > 0
            t̄ = t
        else
            t̲ = t
        end
        t = (t̄ + t̲) / 2
    end

    println(t)
    println()
end

main()

File: problems/chapter06/Ysi/abc026_d.py
from math import sin, pi
def main():
    a, b, c = map(int, input().split())

    def val(t):
        return a * t + b * sin(c * t * pi)

    def is_ok(mid):
        v = val(mid)
        return v > 100
    
    def bisect(ng, ok, error):
        while (error > 10 ** -6):
            mid = (ok + ng) / 2
            if is_ok(mid):
                ok = mid
            else:
                ng = mid
            error = abs(val(ok) - 100)
        return ok

    ans = bisect(0, 100, 100)
    print(ans)
if __name__=='__main__':
    main()